clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -492,4 +533,26 @@ public void onClosed(Consumer<Exception> callback) {
 
         onClosedCallbackList.add(callback);
     }
+
+    private class ConnectionState implements InvocationBinder {
+        HubConnection connection;
+
+        public ConnectionState(HubConnection connection) {
+            this.connection = connection;
+        }
+
+        @Override
+        public Class<?> GetReturnType(String invocationId) {
+            return null;
+        }
+
+        @Override
+        public List<Class<?>> GetParameterTypes(String methodName) throws Exception {
+            List<InvocationHandler> handlers = connection.handlers.get(methodName);
+            if (handlers == null || handlers.size() == 0) {
+                throw new Exception(String.format("There are no callbacks registered for the method '%s'.", methodName));
```

Comment:
Do we throw for this in other clients?

Reply:
https://github.com/aspnet/SignalR/blob/adc5706bcb2b3cb85e98e303999eb9a778426f7c/src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs#L1256

Reply:
That's not the same though right. If there's no method, it logs. If there are no callbacks, it throws.

Reply:
Ok, true

Reply:
Yeah, let's match .NET here